File: BankingTool.Model/Dto/RequestDto/BankAccount/TransactionListRequest.cs
using BankingTool.Model.Dto.RequestDtos;

namespace BankingTool.Model.Dto.RequestDto.BankAccount
{
    public class TransactionListRequestObject
    {
        public TransactionListRequest Request { get; set; }
    }
    public class TransactionListRequest
    {
        public TransactionListRequestAccount Account { get; set; }
        public TransactionListRequestTransaction Transaction { get; set; }
    }
    public class TransactionListRequestAccount : RequestId
    {
        public int? SenderAccountId { get; set; }
        public int? ReceiverAccountId { get; set; }
    }
    public class TransactionListRequestTransaction
    {
        public string TransactionTag { get; set; }
        public DateTime? TransactionFromDate { get; set; }
        public DateTime? TransactionToDate { get; set; }
        public string TransactionCategory { get; set; }
    }
}

File: bankingtool.client/src/app/dto/request/bank-account/transaction-list-request.ts
import { RequestId } from "../request-id";

export interface TransactionListRequestObject {
  request: TransactionListRequest;
}

export interface TransactionListRequest {
  account: TransactionListRequestAccount;
  transaction: TransactionListRequestTransaction;
}

export interface TransactionListRequestAccount extends RequestId {
  senderAccountId: number | null;
  receiverAccountId: number | null;
}

export interface TransactionListRequestTransaction {
  transactionTag: string | null;
  transactionFromDate: Date | null;
  transactionToDate: Date | null;
  transactionCategory: string | null
}

export const createTransactionListRequestObject = (accountId: number, receiverAccountId: number | null, senderAccountId: number | null,
  transactionCategory: string | null, transactionFromDate: Date | null, transactionTag: string | null, transactionToDate: Date | null): TransactionListRequestObject => {
  let model: TransactionListRequestObject = {
    request: {
      account: {
        id: accountId,
        receiverAccountId: receiverAccountId,
        senderAccountId: senderAccountId
      },
      transaction: {
        transactionCategory: transactionCategory,
        transactionFromDate: transactionFromDate,
        transactionTag: transactionTag,
        transactionToDate: transactionToDate
      }
    }
  }
  return model;
}
